src/pip/_internal/cli/autocompletion.py

```diff
@@ -59,6 +59,14 @@ def autocomplete() -> None:
                     print(dist)
                 sys.exit(1)
 
+        should_list_installables = (
+            not current.startswith("-") and subcommand_name == "install"
+        )
+        if should_list_installables:
+            for path in auto_complete_paths(current, "path"):
```

Comment:
Do we want this to be smarter and only complete `.whl`, `.tar.gz`, and directories?

Reply:
Personally I think it’s fine as-is. (Disclaimer: I _never_ use pip’s autocompletion so my opinion on what is useful is close to meaningless.)

Reply:
Yeah, agreed. I realized a smarter completion would also want to include `.egg` files, `.zip`, and who knows what else.

Okay this is ready for review!